### ISSUE TITLE
restore Timeout as a field on InEndpoint

### ISSUE THREAD

**Issue by notnot** [google/gousb#56]
In the Async API issue i read about accidentally hiding the InEndpoint Timeout and the suggestion to expose it again. Please do! I don't know how to exit cleanly when a goroutine is blocked on a Read().

**Comment by zagrodzki** [comment]
Duplicate of #25
